### ISSUE TITLE
[BUG]: All (non-empty) `Map`s and `Set`s are converted to `{}`

### ISSUE THREAD

**Issue by faberchri** [amaabca/sensitive-param-filter#44]
**Describe the bug:**
Passing a `Map` or a `Set` to `SensitiveParamFilter` turns theses data structures into an empty object (printed as `{}`).
I would expect that a `Set` remains unchanged (as it is the case for arrays) and a `Map` is converted into an object with the same keys as the Map and the values filtered if necessary. 

**Sample Reproduction:**
```
  // NOT OK
  const filter = new SensitiveParamFilter()
  console.log(
    filter.filter([
      "map test",
      new Map([
        ["Password", 143534],
        ["345", 2],
      ]),
    ]),
  )
  // Output: [ 'map test', {} ]

  // NOT OK
  console.log(filter.filter(["set test", new Set(["123", "Password"])]))
  // Output: [ 'set test', {} ]

  // OK
  console.log(filter.filter(["array test", ["123", "Password"]]))
  // Output: [ 'array test', [ '123', 'Password' ] ]

  // OK
  console.log(
    filter.filter([
      "object from map test",
      Object.fromEntries(
        new Map([
          ["Password", 134545],
          ["345", 2],
        ]),
      ),
    ]),
  )
  // OUTPUT: [ 'object from map test', { '345': 2, Password: 'FILTERED' } ]
```

**Comment by zjullion** [comment]
Thanks for this bug report! I'm going to see if there are any other built-in JS types that we aren't filtering correctly, and fix those as well. I can't make any promises on a timeline, but this will be fixed.

**Comment by zjullion** [comment]
This bug has been fixed as part of version 1.2.0. As always, the package is available on [npm](https://www.npmjs.com/package/@amaabca/sensitive-param-filter) and [GitHub packages](https://github.com/amaabca/sensitive-param-filter/packages/210545).
